src/ralph/networks/models/networks.py

```diff
@@ -165,23 +165,26 @@ def current_counter_without_model(self):
         Returns:
             counter int
         """
+        start = len(self.hostname_template_prefix)
+        stop = -len(self.hostname_template_postfix)
         hostnames = []
         for model_class in self.HOSTNAME_MODELS:
             item = model_class.objects.filter(
-                hostname__startswith=self.hostname_template_prefix,
-                hostname__endswith=self.hostname_template_postfix
+                hostname__iregex='{}[0-9]+{}'.format(
+                    self.hostname_template_prefix,
+                    self.hostname_template_postfix
+                )
             ).order_by('-hostname').first()
             if item and item.hostname:
-                hostnames.append(
-                    item.hostname[
-                        len(self.hostname_template_prefix):
-                        -len(self.hostname_template_postfix)
-                    ]
-                )
-
+                try:
+                    counter = int(item.hostname[start:stop])
+                except ValueError:
```

Comment:
I think it would be good to add here comment that there can be non integer counters therefore we skip this error.